socket_spec.cr:541 fails in Arch Linux chroot environment

I am trying to build Crystal 0.21.1 and run its tests in a chroot at Arch Linux and get following error:

```
Using /usr/bin/llvm-config [version=3.9.1]
./bin/crystal build  -o .build/all_spec spec/all_spec.cr
Using compiled compiler at .build/crystal
.build/all_spec 
....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................*.........................................................................................................................................................................................................................................................................................................................................................................**................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................*.........................................................................................................................................................................................................................................................**...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................*.....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................*..............................................*................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................Using compiled compiler at .build/crystal
.................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................*.................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................F.....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................***...............................................................................................................................................................................................................................................................................................................................

Pending:
  Code gen: lib codegens lib var set and get
  Code gen: primitives codegens pointer of int
  Code gen: primitives sums two numbers out of an [] of Number
  Semantic: cast casts from union to incompatible union gives error
  Semantic: def overload restricts on generic type with free type arg
  Semantic: def overload restricts on generic type without type arg
  Semantic: instance var doesn't infer type to be nilable if using self.class in call in assign
  Semantic: pointer allows using pointer with subclass
  Semantic: primitives types pointer of int
  Math Functions for computing quotient and remainder
  Thread::ConditionVariable signals
  Thread::ConditionVariable broadcasts
  Thread::ConditionVariable waits and send signal

Failures:

  1) TCPSocket fails when host doesn't exist
     Failure/Error: expect_raises(Socket::Error, /No address found for localhostttttt:12345/) do

       Expected Socket::Error, got #<Errno: connect: No route to host> with backtrace:
         # 0x2369a44: *TCPSocket#initialize<String, Int32, Nil, Nil>:Nil at /build/crystal/src/crystal-0.21.1/src/socket/tcp_socket.cr 71:15
         # 0x2367f15: *TCPSocket#initialize<String, Int32>:Nil at /build/crystal/src/crystal-0.21.1/src/socket/tcp_socket.cr 27:3
         # 0x2367eda: *TCPSocket::new<String, Int32>:TCPSocket at /build/crystal/src/crystal-0.21.1/src/socket/tcp_socket.cr 18:5
         # 0xed451c: ~procProc(Nil) at /build/crystal/src/crystal-0.21.1/src/spec/methods.cr 4:5
         # 0xbb8ff4: *it<String, String, Int32, Int32, &Proc(Nil)>:(Array(Spec::Result) | Nil) at /build/crystal/src/crystal-0.21.1/src/spec/methods.cr 17:7
         # 0xed243d: ~procProc(Nil) at /build/crystal/src/crystal-0.21.1/src/spec/methods.cr 4:5
         # 0x1241de3: *Spec::RootContext::describe<String, String, Int32, &Proc(Nil)>:Spec::Context+ at /build/crystal/src/crystal-0.21.1/src/spec/context.cr 152:30
         # 0xed4c47: *describe<TCPSocket:Class, String, Int32, &Proc(Nil)>:Spec::Context+ at /build/crystal/src/crystal-0.21.1/src/spec/methods.cr 4:5
         # 0xb80b7b: ??? at /build/crystal/src/crystal-0.21.1/spec/std/socket_spec.cr 442:1
         # 0xb974f9: main at /build/crystal/src/crystal-0.21.1/src/main.cr 12:15
         # 0x7efd798b7511: __libc_start_main at ??
         # 0xb7a17a: _start at ??
         # 0x0: ??? at ??

     # spec/std/socket_spec.cr:541

Finished in 7:56 minutes
12124 examples, 1 failures, 0 errors, 13 pending

Failed examples:

crystal spec spec/std/socket_spec.cr:540 # TCPSocket fails when host doesn't exist
make: *** [Makefile:77: spec] Error 1
```

--- a/spec/std/socket_spec.cr
+++ b/spec/std/socket_spec.cr
@@ -538,14 +538,14 @@ describe TCPSocket do
   end
 
   it "fails when host doesn't exist" do
-    expect_raises(Socket::Error, /No address found for localhostttttt:12345/) do
-      TCPSocket.new("localhostttttt", 12345)
+    expect_raises(Socket::Error, /No address found for doesnotexist.example.org.:12345/) do
+      TCPSocket.new("doesnotexist.example.org.", 12345)
     end
   end
 
   it "fails (rather than segfault on darwin) when host doesn't exist and port is 0" do
-    expect_raises(Socket::Error, /No address found for localhostttttt:0/) do
-      TCPSocket.new("localhostttttt", 0)
+    expect_raises(Socket::Error, /No address found for doesnotexist.example.org.:0/) do
+      TCPSocket.new("doesnotexist.example.org.", 0)
     end
   end
 end

